### DISCUSSION TITLE
Branch names are too long for our automation

### DISCUSSION THREAD

**Discussion by maeghan-porter** [renovatebot/renovate#31797] [Request Help] [answered]
### What would you like help with?

Suggestions on how to make the branch names much shorter (a character limit would be nice)

### How are you running Renovate?

Self-Hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Github Actions

### Please tell us more about your question or problem

We are using Renovate to keep our helm releases in Flux up to date. I'm faced with the issue where we have some integration tests that run when a branch is created which spin up an EKS cluster and other resources, which includes a set of ALBs. The networking parts are failing because the branch names are so long and AWS has a restriction of only 32 characters in the names of ALBs and their resources. When we do dev work (which these clusters are typically created for) we use ticket names which are only about 8 characters long. Obviously in the case of renovate staging version updates the branches are named after the helm releases and are obviously typically much longer than the 8 characters we typically use. I got rid of the branch prefix, but outside of that I have no idea how else to shorten these. Am I missing something built-in? Any suggestions to make this work for our use-case? It would be great to have all the tests pass so we can be confident that the version upgrade didn't break anything and just merge it.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment, marked as answer]
https://docs.renovatebot.com/configuration-options/#hashedbranchlength

**Reply by maeghan-porter** [reply]
Amazing, that's exactly what I need, I can't believe my googling for branch name length never yielded this entry in the docs 🤔 
Thank you so much!
